src/common/markdownToHtml.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { PropsWithChildren, useEffect, useState } from 'react'
+import unified from 'unified'
+import remarkParse from 'remark-parse'
+import remark2rehype from 'remark-rehype'
+import raw from 'rehype-raw'
+import stringify from 'rehype-stringify'
+import DOMPurify from 'dompurify'
+
+export default async function convertMarkdownToSanitizedHtml(
+  markdown: string
+): Promise<string> {
+  // remark supSuber library which we use in other places doesn't work with unified
+  // supSuper Issue https://github.com/zestedesavoir/zmarkdown/issues/438
+  const subSup = (mesZL: string): string =>
+    mesZL
+      .replace(/\^([^\^]*)\^/g, '<sup>$1</sup>')
+      .replace(/~([^~]*)~/g, '<sub>$1</sub>')
+
+  // remark uses commonMarkdown, that's why we have to parse italic manually ontop of transforming it with remark
+  // Issue/Question CommonMarkdown Italic https://github.com/remarkjs/remark-rehype/issues/18
+  const italic = (mesZL: string): string =>
+    mesZL.replace(/\*([^*]*)\*/g, '<em>$1</em>')
+
+  //remark sanitize https://github.com/syntax-tree/hast-util-sanitize is not working
+  //we are using DOMPurify instead to sanitize HTML before setting "dangerouslySetInnerHTML"
+
+  const file = await unified()
+    .use(remarkParse)
+    .use(remark2rehype, { allowDangerousHtml: true })
+    .use(raw)
+    .use(stringify)
+    .process(markdown)
+
+  const html = italic(subSup(String(file)))
+  //unified() outputs the HTML within a <p>...</p>, we remove it
+  return DOMPurify.sanitize(html.slice(3, html.length - 4))
+}
+
+type Container = 'div' | 'span'
+interface Props extends PropsWithChildren<any> {
+  container?: Container
+  markdown: string
+}
+export function ContainerWithInnerHtml({
```

Comment:
The name is confusing. The name implies implementation detail, but the component is actually used to render Markdown.

What's the difference between the `Markdown` component? Is it necessary to have three components for rendering Markdown? The new components could replace the existing uses of `ReactMarkdown`.

Reply:
but this is taking care of a mix of html (inline css) and markdown ReactMarkdown deals with regular markdown.

Reply:
i renamed it

Reply:
we have InlineMarkdown and Markdown which are both wrappers around ReactMarkdown adding some custom features. And we now have MarkdownAndHtml which takes Markdown + Html + Inline CSS styles and returns a Html string.

Reply:
Do we need all three? Looks like your implementation can also inline (i.e. `container="span"`). Could be confusing for the users to have two flavours of markdown. E.g. https://www.ebl.lmu.de/corpus/L/99/99 introduction contains `^Um^` which is not formatted.

Reply:
i can merge InlineMarkdown and Markdown in a new PR. But MarkdownAndHtmlToHtml is needed for markdown + html + inline css to html.